### PR TITLE
Add TypeScript definition file

### PR DESCRIPTION
**Pull request by frontweb**
This PR adds TypeScript definition file and updates the README with an example of how to import the function in TypeScript.

Related to https://github.com/auth0/jwt-decode/issues/36

### CODE REVIEW

**Comment by hanssens** [comment]
@frontweb Out of curiosity; any particular reason why manually adding it through @types wouldn't suffice?

```
# see also https://www.npmjs.com/package/@types/jwt-decode
npm install --save-dev @types/jwt-decode
```

Don't get me wrong, being a consumer and TypeScript user I applaud the initiative, but having the .d.ts seperately seems to make more sense in the case where the repo-maintainers don't  intend to update it themselves. 

Also, the [DefinitelyTyped repo for jwt-decode.d.ts](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/types-2.0/jwt-decode) seems to be properly in order, right?

**Comment by frontweb** [comment]
@hanssens my understanding is that bundling the type definitions with the npm package is the preferred method lately.

File: lib/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+declare function jwtDecode(token: string): any;
```
Comment (rebirthtobi) [inline lib/index.d.ts]: this doesn't support the support the second paranmeter `options`

```
    declare module "jwt-decode" {
        interface OptionsType {
            header: boolean;
        }
        type jwtDecodeType = (token: string, options?: OptionsType) => any;
        const jwtDecode: jwtDecodeType;
        export default jwtDecode;
    }
```

Can be modified though

**Comment by frontweb** [comment]
Closing this since types are available through @types/jwt-decode

**Comment by FredericLatour** [comment]
@frontweb @hanssens 

Bundling the type definitions has several advantages:
* less moving parts (you don't need a second package just for the types)
* it makes it more straightforward to keep types in sync.

Even if the maintainers do not write/update types themselves, it is always possible to submit a PR.

**Pull request closed without merging**